NitroxClient/MonoBehaviours/DiscordRP/DiscordController.cs

```diff
@@ -0,0 +1,196 @@
+            else
+            {
+                Log.Info("Discord: Request window is allready active.");
+            }
+        }
+
+        void Update()
+        {
+            DiscordRpc.RunCallbacks();
+        }
+
+        void OnEnable()
+        {
+            Log.Info("Discord: init");
+            handlers = new DiscordRpc.EventHandlers();
+            handlers.readyCallback = ReadyCallback;
+            handlers.disconnectedCallback += DisconnectedCallback;
+            handlers.errorCallback += ErrorCallback;
+            handlers.joinCallback += JoinCallback;
+            handlers.spectateCallback += SpectateCallback;
+            handlers.requestCallback += RequestCallback;
+            DiscordRpc.Initialize(applicationId, ref handlers, true, optionalSteamId);
+
+            ShowingWindow = false;
+        }
+
+        void OnDisable()
+        {
+            Log.Info("Discord: shutdown");
+            DiscordRpc.Shutdown();
+        }
+
+        public void InitDRPDiving(string username, int playercount, string ipAddressPort)
+        {
+            Presence.state = "Diving in game";
+            Presence.details = "Playing as " + username;
+            Presence.startTimestamp = 0;
+            Presence.partyId = username;
+            Presence.partySize = playercount;
+            Presence.partyMax = 99;
+            Presence.joinSecret = CheckIP(ipAddressPort);
+            SendDRP();
+        }
+
+        public void InitDRPMenu()
+        {
+            Presence.state = "In menu";
+            SendDRP();
+        }
+
+        public void UpdateDRPDiving(int playerCount)
+        {
+            Presence.partySize = playerCount;
+            SendDRP();
+        }
+
+        private void SendDRP()
+        {
+            Presence.largeImageKey = "diving";
+            Presence.instance = false;
+            DiscordRpc.UpdatePresence(Presence);
+        }
+
+        public void RespondLastJoinRequest(int accept)
+        {
+            Log.Info("Discord: responding to Join request => " + (accept.Equals("1") ? "true" : "false"));
+            DiscordRpc.Respond(lastJoinRequestUserID, (DiscordRpc.Reply)accept);
+        }
+
+        private GameObject FindObject(GameObject parent, string name)
+        {
+            Component[] trs = parent.GetComponentsInChildren(typeof(Transform), true);
+            foreach (Component t in trs)
+            {
+                if (t.name == name)
+                {
+                    return t.gameObject;
+                }
+            }
+
+            return null;
+        }
+
+        private string CheckIP(string ipPort)
+        {
+            string ip = ipPort.Split(':')[0];
+            string port = ipPort.Split(':')[1];
+
+            if (ip == "127.0.0.1")
+            {
+                WebRequest req = WebRequest.Create("http://checkip.dyndns.org");
+                WebResponse resp = req.GetResponse();
+                StreamReader sr = new StreamReader(resp.GetResponseStream());
```

Comment:
WebClient fails (on my pc) because of certificate issues with mono. We have to figure out something we can rely on.
For now I will use my way in the IpHelper and we should rework it later with a system what not ask a website.